### PR TITLE
chore: Resolve linter issues for unhandled-error

### PR DESCRIPTION
**Pull request by zak-pawel**
Plus some typos etc...

```
internal/process/process.go:84:2       revive       unhandled-error: Unhandled error in call to function p.Stdin.Close
internal/process/process_posix.go:15:3 revive       unhandled-error: Unhandled error in call to function cmd.Process.Signal
internal/process/process_posix.go:20:3 revive       unhandled-error: Unhandled error in call to function cmd.Process.Kill
internal/syslog/framing_test.go:10:2   revive       unhandled-error: Unhandled error in call to function f1.UnmarshalTOML
internal/syslog/framing_test.go:14:2   revive       unhandled-error: Unhandled error in call to function f2.UnmarshalTOML
internal/syslog/framing_test.go:18:2   revive       unhandled-error: Unhandled error in call to function f3.UnmarshalTOML
internal/syslog/framing_test.go:22:2   revive       unhandled-error: Unhandled error in call to function f4.UnmarshalTOML
internal/syslog/framing_test.go:26:2   revive       unhandled-error: Unhandled error in call to function f5.UnmarshalTOML
internal/syslog/framing_test.go:30:2   revive       unhandled-error: Unhandled error in call to function f6.UnmarshalTOML
logger/logger.go:169:3                 revive       unhandled-error: Unhandled error in call to function closer.Close
logger/logger_test.go:144:3            revive       unhandled-error: Unhandled error in call to function w.Write
logger/logger_test.go:19:17            revive       unhandled-error: Unhandled error in call to function os.Remove
logger/logger_test.go:34:17            revive       unhandled-error: Unhandled error in call to function os.Remove
logger/logger_test.go:48:17            revive       unhandled-error: Unhandled error in call to function os.Remove
logger/logger_test.go:63:17            revive       unhandled-error: Unhandled error in call to function os.Remove
logger/logger_test.go:77:17            revive       unhandled-error: Unhandled error in call to function os.Remove
metric/metric.go:253:2                 revive       unhandled-error: Unhandled error in call to function h.Write
metric/metric.go:254:2                 revive       unhandled-error: Unhandled error in call to function h.Write
metric/metric.go:256:3                 revive       unhandled-error: Unhandled error in call to function h.Write
metric/metric.go:257:3                 revive       unhandled-error: Unhandled error in call to function h.Write
metric/metric.go:258:3                 revive       unhandled-error: Unhandled error in call to function h.Write
metric/metric.go:259:3                 revive       unhandled-error: Unhandled error in call to function h.Write
metric/series_grouper.go:100:2         revive       unhandled-error: Unhandled error in call to function mh.WriteByte
metric/series_grouper.go:104:2         revive       unhandled-error: Unhandled error in call to function mh.Write
metric/series_grouper.go:91:2          revive       unhandled-error: Unhandled error in call to function mh.WriteString
metric/series_grouper.go:92:2          revive       unhandled-error: Unhandled error in call to function mh.WriteByte
metric/series_grouper.go:95:3          revive       unhandled-error: Unhandled error in call to function mh.WriteString
metric/series_grouper.go:96:3          revive       unhandled-error: Unhandled error in call to function mh.WriteByte
metric/series_grouper.go:97:3          revive       unhandled-error: Unhandled error in call to function mh.WriteString
metric/series_grouper.go:98:3          revive       unhandled-error: Unhandled error in call to function mh.WriteByte
models/running_output_test.go:43:3     revive       unhandled-error: Unhandled error in call to function ro.Write
models/running_output_test.go:59:4     revive       unhandled-error: Unhandled error in call to function ro.Write
selfstat/selfstat.go:180:2             revive       unhandled-error: Unhandled error in call to function h.Write
selfstat/selfstat.go:191:3             revive       unhandled-error: Unhandled error in call to function h.Write
testutil/testutil_test.go:17:2         revive       unhandled-error: Unhandled error in call to function os.Setenv
testutil/testutil_test.go:25:2         revive       unhandled-error: Unhandled error in call to function os.Setenv
testutil/testutil_test.go:9:2          revive       unhandled-error: Unhandled error in call to function os.Unsetenv
           
```

### CODE REVIEW

File: cmd/telegraf/main.go [hunk 1]
```diff
@@ -118,7 +119,10 @@ func runApp(args []string, outputBuffer io.Writer, pprof Server, c TelegrafConfi
 
 	// This function is used when Telegraf is run with only flags
 	action := func(cCtx *cli.Context) error {
-		logger.SetupLogging(logger.LogConfig{})
+		err := logger.SetupLogging(logger.LogConfig{})
+		if err != nil {
+			return fmt.Errorf("E! %w", err)
+		}
```
Comment (powersj) [inline cmd/telegraf/main.go:125]: I already wonder what errors this is going to expose that users have not seen :)

File: cmd/telegraf/telegraf_windows.go [hunk 2]
```diff
@@ -135,7 +136,11 @@ func (t *Telegraf) runAsWindowsService() error {
 			return fmt.Errorf("E! " + err.Error())
 		}
 	} else {
-		logger.SetupLogging(logger.LogConfig{LogTarget: logger.LogTargetEventlog})
+		err = logger.SetupLogging(logger.LogConfig{LogTarget: logger.LogTargetEventlog})
+		if err != nil {
+			return fmt.Errorf("E! " + err.Error())
```
Comment (sspaink) [inline cmd/telegraf/telegraf_windows.go:141]:
```suggestion
			return err.Error()
```

This `E!` should also be added by main

**Review by powersj — APPROVED** [review]
I'm good with the changes, I do have some concern on exposing errors suddenly that previously may not have, but with logging I hope (?!) that the exposure is limited.

Curious what others on the team think.

File: plugins/outputs/postgresql/sqltemplate/template.go [hunk 3]
```diff
@@ -1,9 +1,10 @@
+// Package sqltemplate /*
```
Comment (sspaink) [inline plugins/outputs/postgresql/sqltemplate/template.go:1]:
```suggestion
// Package sqltemplate
```

File: cmd/telegraf/main.go (hunk 1 above)
Comment (zak-pawel) [reply cmd/telegraf/main.go:125]: Errors regarding `close` problems for previous logger when new logger will be set up using this method...
Is that ok? Or should we swallow this error and not return it?

File: cmd/telegraf/main.go [hunk 4]
```diff
@@ -118,7 +119,10 @@ func runApp(args []string, outputBuffer io.Writer, pprof Server, c TelegrafConfi
 
 	// This function is used when Telegraf is run with only flags
 	action := func(cCtx *cli.Context) error {
-		logger.SetupLogging(logger.LogConfig{})
+		err := logger.SetupLogging(logger.LogConfig{})
+		if err != nil {
+			return fmt.Errorf("E! %w", err)
```
[the remark this answers is not in the document]
Comment (zak-pawel) [reply cmd/telegraf/main.go:124]: @sspaink Should I do the same with every single `return fmt.Errorf("E! ... ")` inside `action := func(cCtx *cli.Context) error {`?

File: cmd/telegraf/telegraf_windows.go (hunk 2 above)
Comment (zak-pawel) [reply cmd/telegraf/telegraf_windows.go:141]: @sspaink Should I do the same with every single `return fmt.Errorf("E! ... ")` inside `func (t *Telegraf) runAsWindowsService() error {`?

File: cmd/telegraf/telegraf.go [hunk 5]
```diff
@@ -187,18 +188,18 @@ func (t *Telegraf) runAgent(ctx context.Context) error {
 	}
 
 	if !(t.test || t.testWait != 0) && len(c.Outputs) == 0 {
-		return errors.New("Error: no outputs found, did you provide a valid config file?")
+		return errors.New("error: no outputs found, did you provide a valid config file?")
```
[the remark this answers is not in the document]
Comment (zak-pawel) [reply cmd/telegraf/telegraf.go]: Done

File: cmd/telegraf/telegraf.go [hunk 6]
```diff
@@ -187,18 +188,18 @@ func (t *Telegraf) runAgent(ctx context.Context) error {
...
 	}
 	if t.plugindDir == "" && len(c.Inputs) == 0 {
-		return errors.New("Error: no inputs found, did you provide a valid config file?")
+		return errors.New("error: no inputs found, did you provide a valid config file?")
```
[the remark this answers is not in the document]
Comment (zak-pawel) [reply cmd/telegraf/telegraf.go]: Done

File: plugins/outputs/postgresql/sqltemplate/template.go (hunk 3 above)
Comment (zak-pawel) [reply plugins/outputs/postgresql/sqltemplate/template.go]: Done

File: cmd/telegraf/main.go (hunk 4 above)
[the remark this answers is not in the document]
Comment (sspaink) [reply cmd/telegraf/main.go:124]:
```suggestion
			return err
```
Having the other errors match without `E!` would be a good idea.

[the remark this answers is not in the document]
Comment (zak-pawel) [reply cmd/telegraf/main.go]: @sspaink Done

File: cmd/telegraf/telegraf_windows.go (hunk 2 above)
Comment (zak-pawel) [reply cmd/telegraf/telegraf_windows.go]: @sspaink Done

**Review by sspaink — APPROVED** [review]

**Review by powersj — APPROVED** [review]

**Pull request merged**